common/src/types.rs

```diff
@@ -477,7 +710,19 @@ impl<'a> Name<'a> {
         }
     }
 
-    /// Constructs a new mangled symbol with known language.
+    /// Constructs a new mangled name with a known [`Language`].
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use symbolic_common::{Language, Name};
+    ///
+    /// let name = Name::with_language("_ZN3foo3barEv", Language::Cpp);
+    /// println!("{:#}", name); // prints "_ZN3foo3barEv [C++]"
+    /// # assert_eq!(format!("{:#}", name), "_ZN3foo3barEv [C++]");
```

Comment:
I would personally prefer to just have the `assert_eq` call shown there. Similarly for the `Language` and `Arch` examples.